Updated Guide on defining types (markdown) (#6)
oppia/oppia-web-developer-docs: Guide-on-defining-types.md

      fingerprint.push([kGramHashes[hashIndex], hashIndex]);
});
```

### [TS-9]:
 The operand of a 'delete' operator must be optional.
#### [TS-9-1]:
##### Violation:

> This error occurs when using the delete operator in strictNullChecks. Because the property is no longer implemented after deletion, set its type to undefined. As a result, the property must be undefined; otherwise, the delete operator leads to an error.

Reviewer: ```suggestion
This error occurs when using the `delete` operator with strictNullChecks. Since a property cannot be implemented after deletion, its type is set to undefined. So, the property must contain undefined as a type, otherwise the delete operator leads to an error.
```

Author: Done

```typescript
export class OutcomeDestinationEditorComponent implements OnInit {
  ...
  outcomeNewStateName: string;
  ...
  ngOnInit(): void {
  ...
  delete this.outcomeNewStateName; // error
  ...
}
```

##### Solution:

> One of the potential fix to this error is to assign `undefined` but generally we try to avoid unnecessary use of `undefined`. Because usage of `undefined` does not represent or give proper information about the actual situation.
> 
> As a result, in this case, we choose to use `null` rather than `undefined`. Because the delete operator makes a property unusable, `null` is a better representation of emptiness, unusable or property is now no longer implemented.

Reviewer: ```suggestion
One of the potential fixes to this error is to assign `undefined` as a type but we generally try to avoid it's usage because it does not represent or give proper information about the actual situation.

So, in this case, we choose to use `null` rather than `undefined`. Since the delete operator makes a property unusable, `null` is a better representation of emptiness or unusability and property is now no longer implemented.
```

Author: Done
